Api/Services/IngredientsService.cs

```diff
@@ -98,6 +99,115 @@ public async Task<Result<IngredientVM>> CreateIngredientAsync(CreateIngredientRe
         };
     }
 
+    /// <summary>
+    /// Gets the change history of an ingredient with optional filters and pagination.
+    /// </summary>
+    /// <param name="request">Request with parameters</param>
+    /// <param name="ingredientId">Ingredient id</param>
+    /// <param name="userId">ID of the current user</param>
+    /// <param name="page">Page number</param>
+    /// <param name="perPage">Records per page</param>
+    /// <returns>Paginated list of ingredient amount corrections</returns>
+    [ErrorCode(null, ErrorCodes.NotFound)]
+    [MethodErrorCodes<AuthorizationService>(nameof(AuthorizationService.VerifyRestaurantBackdoorAccess))]
+    public async Task<Result<Pagination<IngredientAmountCorrectionVM>>> GetIngredientHistoryAsync(
+        IngredientHistoryRequest request,
+        int ingredientId,
+        Guid userId,
+        int page, 
+        int perPage
+)
+    {
+        // Check if the ingredient exists
+        var ingredient = await dbContext.Ingredients
+            .Include(i => i.MenuItems)
+            .ThenInclude(mi => mi.MenuItem)
+            .FirstOrDefaultAsync(i => i.IngredientId == ingredientId);
+        
+        if (ingredient == null)
+        {
+            return new ValidationFailure
+            {
+                ErrorCode = ErrorCodes.NotFound,
+                ErrorMessage = ErrorCodes.NotFound,
+                PropertyName = nameof(ingredientId)
```

Comment:
```suggestion
                PropertyName = null,
```
jak napisałeś w atrybucie